src/unix/pipe.c

```diff
@@ -457,25 +461,30 @@ int uv_pipe_chmod(uv_pipe_t* handle, int mode) {
     return UV_EINVAL;
 
   /* Unfortunately fchmod does not work on all platforms, we will use chmod. */
-  name_len = 0;
-  r = uv_pipe_getsockname(handle, NULL, &name_len);
-  if (r != UV_ENOBUFS)
+  name_len = sizeof(scratch);
+  r = uv_pipe_getsockname(handle, scratch, &name_len);
+  if (r != 0 && r != UV_ENOBUFS)
```

Comment:
This can't realistically happen because the output is at most `sockaddr_un.sun_path` long (100-150 bytes)

For the same reason, the alloc-and-retry path below is likely dead code.

Reply:
That will simplify things!

Reply:
Done!